src/Symfony/Component/Security/Http/EventListener/AuthenticatingListener.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Symfony\Component\Security\Http\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+use Symfony\Component\Security\Core\Exception\LogicException;
+use Symfony\Component\Security\Http\Authenticator\CustomAuthenticatedInterface;
+use Symfony\Component\Security\Http\Authenticator\PasswordAuthenticatedInterface;
+use Symfony\Component\Security\Http\Authenticator\TokenAuthenticatedInterface;
+use Symfony\Component\Security\Http\Event\VerifyAuthenticatorCredentialsEvent;
+
+/**
+ * This listeners uses the interfaces of authenticators to
+ * determine how to check credentials.
+ *
+ * @author Wouter de Jong <wouter@driveamber.com>
+ *
+ * @final
+ * @experimental in 5.1
+ */
+class AuthenticatingListener implements EventSubscriberInterface
+{
+    private $encoderFactory;
+
+    public function __construct(EncoderFactoryInterface $encoderFactory)
+    {
+        $this->encoderFactory = $encoderFactory;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [VerifyAuthenticatorCredentialsEvent::class => ['onAuthenticating', 128]];
```

Comment:
Any reason why so high? This almost feels like it should be negative - that you could put a listener in *front* of this (for some reason) to "replace" it.

Reply:
Btw, maybe call this class `VerifyAuthenticatorCredentialsListener`?

Also, `encoderFactory` should be injected via a service locator for laziness (as my authenticator my not need it) or this should be split into 3 different listeners for the 3 different cases.

Reply:
Yes, we definitely need to list all Security events and find duplicates. I.e. there are 2 failure events and I believe also 2 success events.

Reply:
> Also, encoderFactory should be injected via a service locator for laziness (as my authenticator my not need it) or this should be split into 3 different listeners for the 3 different cases.

Is there any problem with injecting `encoderFactory` non-lazy? Seems like it's not a very heavy service to me. Splitting this into 3 different listeners doesn't solve the issue with lazyness - it would still be injected - and I put it in one (although it's not that nice) to ease triggering the exception when no credentials-interface was implemented.

Reply:
You’re right about the 3 listeners. About the laziness part, it might be light enough. I guess since this will be triggered rarely (it’s not an even that will be triggered on each request), it’s probably ok.